dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityCacheTest.java

```diff
@@ -129,6 +146,31 @@ void shouldFindTrackedEntityWhenAnotherOwnerAlreadyInCache()
     Assertions.assertContainsOnly(List.of(trackedEntityC), trackedEntities);
   }
 
+  @Test
+  void shouldHaveAccessAfterEnrollmentPersistedAndOwnershipCreatedAndCacheInvalidated() {
+    TrackedEntity trackedEntity = manager.get(TrackedEntity.class, "H8732208127");
+    assertNotNull(trackedEntity);
+    User user = manager.get(User.class, "lPaILkLkgOM");
+
+    injectSecurityContextUser(user);
+
+    assertFalse(
```

Comment:
Maybe I would add some comments to describe what is happening with the cache, when it is populated and when it is invalidated.
Cache population/invalidation is a side effect so it is not easy to understand looking at the calls here.